File: task9/src/classesFigure/base/Polygon.java
package classesFigure.base;
import interfaceFigure.Perimeter;


/**
 * perimeter - хранит периметр
 * calculationPerimeter - вычисляет периметр складывая параметры сторон объекта
 */
public abstract class Polygon extends Figure implements Perimeter {
    protected int perimeter;
    public int getPerimeter() {
        return this.perimeter;
    }
    protected int calculationPerimeter() {
        int sum = 0;
        for (int p : super.parameters) {
            sum += p;
        }
        return sum;
    }
}

File: task13/base/Polygon.py
from base.Figure import Figure
from interface.Perimeter import Perimeter

class Polygon(Figure, Perimeter):

    def __init__(self, name, square, perimeter):
        super().__init__(name, square)
        self.__perimeter = perimeter

    def get_perimeter(self):
        return self.__perimeter

    def get_info(self):
        return f"{super().get_info()}, периметер: {self.__perimeter}"